Add documentation for alter_data_node (#1834)
timescale/docs: api/alter_data_node.md

### Returns

|Column|Description|
|-|-|
|`node_name`|Local name to use for the data node|
|`host`|Host name for the remote data node|
|`port`|Port for the remote data node|
|`database`|Database name used on the remote data node|
|`available`|Availability of the remote data node for read/write queries|

#### Errors

An error is given if:

*   A remote data node with the provided `node_name` argument does not exist.

#### Privileges

To alter a data node, you must have the correct permissions, or be the owner of the remote server.
Additionally, you must have the `USAGE` privilege on the `timescaledb_fdw` foreign data
wrapper.

### Sample usage

To change the port number and host information for an existing data node `dn1`:

```sql
SELECT alter_data_node('dn1', host => 'dn1.example.com', port => 6999);
```

Data nodes are available for read/write queries by default. If the data node
becomes unavailable for some reason, the read/write query gives an error. This
API provides an optional argument, `available`, to mark an existing data node
as available or unavailable for read/write queries. By marking a data node as

> unavailable you can allow read/write queries to proceed in the cluster as
> described in the [multi-node HA section][multi-node-ha]

Reviewer: ```suggestion
unavailable you can allow read/write queries to proceed in the cluster. 
For more information, see the [multi-node HA section][multi-node-ha].
```

Author: Done.

### Returns

| Column      | Description                                                 |
|-------------|-------------------------------------------------------------|
| `node_name` | Local name to use for the data node                         |
| `host`      | Host name for the remote data node                          |
| `port`      | Port for the remote data node                               |
| `database`  | Database name used on the remote data node                  |
| `available` | Availability of the remote data node for read/write queries |

#### Errors

An error is given if:

*   A remote data node with the provided `node_name` argument does not exist.

#### Privileges

To alter a data node, you must be a superuser or must be owner of the remote server.
Additionally, you must have the `USAGE` privilege on the `timescaledb_fdw` foreign data
wrapper.

### Sample usage

To change the port number and host information for an existing data node `dn1`:

```sql
SELECT alter_data_node('dn1', host => 'dn1.example.com', port => 6999);
```

> By default a data node is available for read/write queries. Reads/writes start erroring
> out if the data node becomes unavailable for whatever reasons. This api provides an

Reviewer: ```suggestion
Data nodes are available for read/write queries by default. If the data 
node becomes unavailable for some reason, the read/write query gives 
an error. This API provides an
```

> optional argument `available` to mark an existing data node as available/unavailable

Reviewer: ```suggestion
optional argument, `available`, to mark an existing data node as available or unavailable
```
